shared/AppInsightsCore/src/JavaScriptSDK/HelperFuncs.ts

```diff
@@ -44,17 +109,17 @@ export function isNotNullOrUndefined(value: any): boolean {
 }
 
 export function hasOwnProperty(obj: any, prop: string): boolean {
-    return obj && ObjHasOwnProperty.call(obj, prop);
+    return !!(obj && ObjHasOwnProperty.call(obj, prop));
```

Comment:
The !! is to ensure that if undefined or null is passed the return is still a boolean value (false in both of those cases)